core/src/main/scala/org/apache/spark/scheduler/ExecutorFailuresInTaskSet.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.scheduler
+
+import scala.collection.mutable.HashMap
+
+/**
+ * Small helper for tracking failed tasks for blacklisting purposes.  Info on all failures on one
+ * executor, within one task set.
+ */
+private[scheduler] class ExecutorFailuresInTaskSet(val node: String) {
+  /**
+   * Mapping from index of the tasks in the taskset, to the number of times it has failed on this
+   * executor and the expiry time.
+   */
+  val taskToFailureCountAndExpiryTime = HashMap[Int, (Int, Long)]()
```

Comment:
its fine, I can remove it here.  I just pushed a commit removing most of it (https://github.com/apache/spark/pull/15249/commits/354f36bd36c7615883c08542eea333704e421164).  I left in some of the stuff in `BlacklistTracker` around interaction w/ the legacy conf, just pending the ongoing discussion, but I can update that as well.

`ExecutorFailuresInTaskSet` now seems like a bit of overkill, but hopefully its ok if we keep that for now -- I'll update the pr description a bit.